src/Simplify.cpp

```diff
@@ -1449,7 +1449,9 @@ class Simplify : public IRMutator {
         } else if (is_one(b)) {
             expr = a;
         } else if (equal(a, b) &&
-                   !is_zero(b)) {
+                   !is_zero(b) &&
+                   // (n/0)/(n/0) should not be simplified to 1
+                   !(div_a && is_zero(div_a->b))) {
```

Comment:
Maybe `signed_integer_overflow` should just be generalized. Is there any time where we might care something is signed integer overflow, but not division by zero, or vice versa?